### PR TITLE
FCM configuration

### PR DESCRIPTION
**Pull request by waelba**
- Support FCM
- Comment GCM

### CODE REVIEW

**Comment by DrissTM** [comment]
AFAICS it seems good. I'll perform some tests and will let you know ;)

**Review by DrissTM — APPROVED** [review]
I've tested all the cases I could think of:
- With chat view displayed (no notification expected)
- While being in the app but not in the chat view
- With the app in background
- With the app killed

And everything is working great 👍

File: build.gradle [hunk 1]
```diff
@@ -6,12 +6,14 @@ buildscript {
...
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:2.3.0'
+        classpath 'com.google.gms:google-services:3.1.1'
     }
 }
 
 allprojects {
     repositories {
         jcenter()
+        maven { url "https://maven.google.com" }
```
Comment (jarthod) [inline build.gradle:17]: why maven?

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Project-wide Gradle settings.
 
-GCM_API_KEY="change here"
-DIMELO_SDK_SECRET="change here"
\ No newline at end of file
+#GCM_API_KEY="change here"
```
Comment (jarthod) [inline gradle.properties:3]: no firebase key?

File: build.gradle (hunk 1 above)
Comment (waelba) [reply build.gradle:17]: It's requested by FCM compile: https://github.com/firebase/quickstart-android/blob/a9227f0b8cbf941402bc6b3fd90ab449e8be9e56/messaging/build.gradle#L23

File: gradle.properties (hunk 2 above)
Comment (waelba) [reply gradle.properties:3]: Yes this line for GCM implementation. For FCM we use `google-services.json`

Comment (jarthod) [reply gradle.properties:3]: Ok

File: build.gradle (hunk 1 above)
Comment (jarthod) [reply build.gradle:17]: ok

**Review by jarthod — APPROVED** [review]

**Pull request merged**